### PR TITLE
Scripts: Only use svgr/webpack in js files

### LINKED ISSUE #34376: Relative image urls in @wordpress/scripts 18.0.0
### Description

Hey, In @wordpress/scripts 18.0.0 I understand that there is breaking changes.

I’m using this for compiling both js and scss. And I now get `TypeError: url.replace is not a function` when trying to compile a scss file that contains relative paths to image files, like `background-image: url('../images/image.svg');`

Thought it was because of me having a bit of a custom webpack config. But also failed with a fresly made @wordpress/create-block

### Step-by-step reproduction instructions

1. `npx @wordpress/create-block todo-list`
2. Add an svg to /todo-list/images
3. Add `background-image: url(../images/yourimage.svg);` to style.scss
4. `npm run build`

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Pre-checks

- [X] I have searched the existing issues.
- [X] I have tested with all plugins deactivated except Gutenberg.

### PR DESCRIPTION
**Pull request by walbo**
## Description
Only use svgr/webpack loader on js files and inline svg's from css files.

Fixes #34376 

## How has this been tested?
1. In this branch: `npx wp-create-block example --no-wp-scripts --template wp-block-directory-template`
2. `cd example`
3. Created a svg files and used it in both a js file and as a background image in a scss file.
4. `../node_modules/.bin/wp-scripts build` 
5. No errors and checked build files that there are inlined.

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/scripts/config/webpack.config.js [hunk 1]
```diff
@@ -184,9 +184,15 @@ const config = {
...
 			{
 				test: /\.svg$/,
+				issuer: /\.jsx?$/,
 				use: [ '@svgr/webpack', 'url-loader' ],
 				type: 'javascript/auto',
 			},
+			{
+				test: /\.svg$/,
+				issuer: /\.(sc|sa|c)ss$/,
+				type: 'asset/inline',
```
Comment (walbo) [inline packages/scripts/config/webpack.config.js:194]: In `wp-scripts@17` the svg files was inlined so `asset/inline` is used there, but should this be changed to `asset/resource` and move the svgs to the `images` folder?

Comment (gziolo) [reply packages/scripts/config/webpack.config.js:194]: That's interesting. So we can have now different handling in CSS and JS, awesome.

We should check how it worked with `@wordpress/scripts` v17 that was using webpack 4 and provide the same experience for backward compatibility.

Comment (gziolo) [reply packages/scripts/config/webpack.config.js:194]: https://unpkg.com/browse/@wordpress/scripts@17.0.0/config/webpack.config.js

It was as simple as:

```js
{
    test: /\.svg$/,
    use: [ '@svgr/webpack', 'url-loader' ],
},
```

Comment (gziolo) [reply packages/scripts/config/webpack.config.js:194]: Do you think we can add `svg` to the resource handler for other images like gif or png so they have consistent processing?

Comment (walbo) [reply packages/scripts/config/webpack.config.js:194]: > Do you think we can add `svg` to the resource handler for other images like gif or png so they have consistent processing?

Yes, we can do that. That will create a svg file in the `images` folder and not inline them like `url-loader` did in v17. However I think that is fine, and personally I would prefer it. Both approaches has pros and cons.

What do you think @gziolo. Inline them or move svgs to the resouce handler?

Comment (gziolo) [reply packages/scripts/config/webpack.config.js:194]: I see now that the default was no limit, so it was inlining all SVG files. Processing all the changes in webpack 5 API is a mountain of work. 

Let me test the current setup with some custom plugin and I will report back 😄

Comment (walbo) [reply packages/scripts/config/webpack.config.js:194]: We could use the general `asset` type, where webpack will automatically choose between `resource` and `inline` based on filesize.

See https://webpack.js.org/guides/asset-modules/#general-asset-type

Comment (gziolo) [reply packages/scripts/config/webpack.config.js:194]: This page is helpful. Let's try back compact version first. We can iterate further 😄 

On the bright side, the refactoring they applied makes a ton of sense ❤️

**Review by gziolo — APPROVED** [review]
Everything works correctly with the updated configuration 👍🏻 

I included a changelog entry.

**Pull request merged**